elfi/distributions.py

```diff
@@ -32,6 +32,10 @@ def pdf(self, x, *params, **kwargs):
     def logpdf(self, x, *params, **kwargs):
         raise NotImplementedError
 
+    @property
+    def name(self):
+        return self._name or self.__class__.__name__
+
 
```

Comment:
I was thinking of a more permissive strategy with duck typing. The only requirement being, that they implement the `rvs` and `pdf` methods. Then we can for example directly use `rv_continuous` distributions without wrapping them, and also it doesn't matter if they are instances or just static classes.